### ISSUE TITLE
Enhance forecasting with more features

### ISSUE THREAD

**Issue by americast** [georgia-tech-db/evadb#1243]
### Search before asking

- [X] I have searched the EvaDB [issues](https://github.com/georgia-tech-db/eva/issues) and found no similar feature requests.

### Description

It would be nice to have the following additional features for EvaDB:
1. Add GPU support
2. Static exogenous variables
3. Multivariate forecasting
4. Error when there are <=1 data points
5. Explain `unique_id` better
6. catch frequency related errors better

### Use case

1. GPU support is essential in order to perform autoML techniques quickly. Currently, it has been disabled owing to instability with `ray`.
2. Right now, every exogenous variable is considered temporal in nature. It is important to distinguish them from static ones.
3. Right now, we can use `PREDICT` in only one column. It would be great to be able to predict forecasting for multiple columns at once.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by khushitalesra** [comment]
If the price column has comma then forecasting doesn't work. Below error is encountered
ValueError: could not convert string to float: '270,000'
EvaDB Session ended with an error: could not convert string to float: '270,000'

**Comment by xzdandy** [comment]
@gaurav274 LLM based data cleaning can help this case. In this case, without LLM it is also possible, but I think LLM is more general and flexible. One optimization is to only run LLM data cleaning on the tuple that failed instead of all tuples, which saves the cost and time. 

An alternative approach is to skip the rows that does not match the type.
@khushitalesra Is the price column type float? But the content is string in this case.

**Comment by khushitalesra** [comment]
yes earlier I had defined the price column to string because of the comma but then I changed it to float as it was throwing error

**Comment by gaurav274** [comment]
> @gaurav274 LLM based data cleaning can help this case. In this case, without LLM it is also possible(e.g., REGEX), but I think LLM is more general and flexible. One optimization is to only run LLM data cleaning on the tuple that failed instead of all tuples, which saves the cost and time. Another optimization is to choose the LLM model, for a simple task like this we may use a lightweight local model.
> 
> An alternative approach is to skip the rows that does not match the type. @khushitalesra Is the price column type float? But the content is string in this case.

@xzdandy This is a nice application of format inconsistencies.

**Comment by gaurav274** [comment]
@americast let us break this into sub-tasks and assign a separate issue to each. Thanks!

**Comment by americast** [comment]
> If the price column has comma then forecasting doesn't work. Below error is encountered ValueError: could not convert string to float: '270,000' EvaDB Session ended with an error: could not convert string to float: '270,000'

Fixed in #1283
